radixdlt-java/src/main/java/com/radixdlt/client/application/translate/data/SendMessageToParticlesMapper.java

```diff
@@ -15,52 +17,64 @@
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Function;
 import java.util.function.Supplier;
+import java.util.stream.Stream;
 
 /**
  * Maps a send message action to the particles necessary to be included in an atom.
  */
 public class SendMessageToParticlesMapper implements ActionToParticlesMapper {
+
+	/**
+	 * A module capable of creating new securely random ECKeyPairs
+	 */
 	private final Supplier<ECKeyPair> generator;
 
+	/**
+	 * Function describing given an action who do we encrypt a message for
+	 */
+	private final Function<SendMessageAction, Stream<ECPublicKey>> encryptionScheme;
+
 	public SendMessageToParticlesMapper(Supplier<ECKeyPair> generator) {
+		this(generator, sendMsg -> Stream.of(sendMsg.getFrom(), sendMsg.getTo()).map(RadixAddress::getPublicKey));
+	}
+
+	public SendMessageToParticlesMapper(Supplier<ECKeyPair> generator, Function<SendMessageAction, Stream<ECPublicKey>> encryptionScheme) {
 		this.generator = generator;
+		this.encryptionScheme = encryptionScheme;
 	}
 
+	/**
+	 * If SendMessageAction is unencrypted, returns a single message particle containing the
+	 * payload data.
+	 *
+	 * If SendMessageAction is encrypted, creates a private key encrypted by both from and to
+	 * users, stores that into a message particles and then creates another message particle
+	 * with the payload encrypted by the newly created private key.
+	 *
+	 * @param action the action to map to particles
+	 * @return an observable particles of a SendMessageAction. Otherwise, an empty observable
+	 */
```

Comment:
Maybe this description would benefit from one more iteration.

* It's a bit unclear what `SendMessageAction` refers to. Is it the provided `Action` argument?
* The description seems to focus a bit too much on "how" rather than "what".

There are a couple of recommendations in above resource regarding this as well ("Implementation-independence", "First sentence" etc)